pkg/inject/inject.go

```diff
@@ -177,7 +178,11 @@ func (conf *ResourceConfig) AppendPodAnnotation(k, v string) {
 
 // YamlMarshalObj returns the yaml for the workload in conf
 func (conf *ResourceConfig) YamlMarshalObj() ([]byte, error) {
-	return yaml.Marshal(conf.workload.obj)
+	j, err := jsonfilter.Marshal(conf.workload.obj)
```

Comment:
This should definitely have a comment explaining why we're not using `encoding/json` here.